nipoppy_cli/nipoppy/workflows/pipeline.py

```diff
@@ -124,19 +125,63 @@ def pipeline_config(self) -> ProcPipelineConfig:
     def fpath_container(self) -> Path:
         """Return the full path to the pipeline's container."""
         fpath_container = self.pipeline_config.get_fpath_container()
+
         if fpath_container is None:
             raise RuntimeError(
                 f"No container image file specified in config for pipeline"
                 f" {self.pipeline_name} {self.pipeline_version}"
             )
 
-        elif not fpath_container.exists():
-            raise FileNotFoundError(
-                f"No container image file found at {fpath_container} for pipeline"
-                f" {self.pipeline_name} {self.pipeline_version}"
-            )
+        elif not fpath_container.exists(follow_symlinks=True):
+
+            if not is_datalad_installed() or not self.is_datalad_dataset:
+                raise FileNotFoundError(
+                    f"No container image file found at {fpath_container} for pipeline"
+                    f" {self.pipeline_name} {self.pipeline_version}"
+                )
+
+            # try:
+            #     uri = self.pipeline_config.CONTAINER_INFO.URI
+            # except Exception:
+            #     uri = None
+            # if uri is not None:
+            #     from datalad_container import containers_add
+            #     adder = containers_add.ContainersAdd()
+            #     self.logger.info(
+            #         "Adding container image for "
+            #         f"{self.pipeline_name} {self.pipeline_version} "
+            #         f"at {self.dpath_root / fpath_container}")
+            #     adder(name=self.pipeline_name, url=uri, image=fpath_container)
+
+            #     if not fpath_container.exists():
+            #          raise RuntimeError(
+            #               f"Could not add image for pipeline"
+            #               f" {self.pipeline_name} {self.pipeline_version}"
+            #           )
```

Comment:
probably just remove for now and re-add if some users want to add containers that are not in those maintained by repronim

Reply:
heudiconv is included in the repronim containers

will remove this code for now and can be implemented later if requested